spec/lib/tasks/existing_firms_sign_up_task_spec.rb

```diff
@@ -1,107 +1,117 @@
+class StubInvitationHelper
+  def invitation_url(user)
+    "invitation for user #{user.id}"
+  end
+end
+
 module Tasks
   RSpec.describe ExistingFirmsSignUpTask do
     describe '#notify' do
-      let!(:bills_firm) { create :firm }
-      let!(:bill) { create :principal, email_address: 'bill@example.com', firm: bills_firm }
+      let!(:bills_firm) { create :firm, registered_name: 'Wright, Johnston & MacKenzie LLP' }
+      let!(:bill) do
+        principal = build :principal
+        principal.email_address =  'bill@example.com'
+        principal.firm = bills_firm
+        principal.first_name = 'Bill'
+        principal.last_name = 'Junior, the third'
+        principal.save!
+        principal
+      end
+      let(:stub_inviter) { StubInvitationHelper.new }
+      let(:output) { [] }
+
+      it 'invites principals that have no account' do
+        expect(User.count).to eq(0)
+        described_class.notify(stub_inviter, output)
+        expect(User.first).to be_invited_to_sign_up
+      end
+
+      it 'sets user.invitation_sent_at' do
+        expected_invitation_sent_at = DateTime.current
+        allow(DateTime).to receive(:current).and_return(expected_invitation_sent_at)
 
-      before do
+        described_class.notify(stub_inviter, output)
+        expect(User.first.invitation_sent_at.to_datetime).to be_within(1.second).of(expected_invitation_sent_at)
+      end
+
+      it 'invites but does not send email' do
         ActionMailer::Base.deliveries = []
+        described_class.notify(stub_inviter, output)
+        expect(ActionMailer::Base.deliveries).to be_empty
       end
 
-      it 'mails principals that have no account' do
-        create :principal, email_address: 'ben@example.com', firm: create(:firm)
+      context 'creates a csv entry for principal' do
+        before do
+          described_class.notify(stub_inviter, output)
+        end
 
-        described_class.notify
+        it 'frn' do
```

Comment:
The `it` blocks here and the containing `context` block aren't named according to their conventions. See here for an example for describing `context` blocks: http://betterspecs.org/#contexts (generally 'when' or 'with' are the keywords to look for — though you could rephrase this example in a few ways using `describe` too)